skymp5-server/cpp/server_guest_lib/CropHealthRegeneration.cpp

```diff
@@ -1,12 +1,29 @@
 #include "CropHealthRegeneration.h"
+#include "GetBaseActorValues.h"
+#include "MpActor.h"
+
+namespace {
+float PercentToFloat(float percent)
+{
+  return percent / 100.0f;
+}
+}
+
 float CropHealthRegeneration(float newDamageModifier,
                              float secondsAfterLastRegen, MpActor* actor)
 {
-  if (newDamageModifier >= 0.0f && newDamageModifier <= 1.0f) {
-    return newDamageModifier;
-  } else if (newDamageModifier >= 1.0f) {
-    return 1.0f;
-  } else {
-    return 0.0f;
-  }
+  uint32_t baseId = actor->GetBaseId();
+  auto look = actor->GetLook();
+  uint32_t raceId = look ? look->raceId : 0;
+  BaseActorValues baseValues = GetBaseActorValues(baseId, raceId);
+
+  float validHealthRegenerationPercentage =
+    PercentToFloat(baseValues.healRate) *
+    PercentToFloat(baseValues.healRateMult) * secondsAfterLastRegen;
+
+  if (newDamageModifier > validHealthRegenerationPercentage)
+    return validHealthRegenerationPercentage;
+  if (newDamageModifier < 0)
+    return 0;
```

Comment:
Oops, that's not right actually, add brackets please to comply to guidelines

https://github.com/skyrim-multiplayer/skymp/blame/43d7b542d99b0b45ee56c339f143f16657c926c9/docs/docs_cpp_code_guidelines.md#L9-L17